lib/net/ldap/connection.rb

```diff
@@ -52,7 +52,8 @@ def open_connection(server)
     hosts.each do |host, port|
       begin
         prepare_socket(server.merge(socket: @socket_class.new(host, port, socket_opts)), timeout)
-        if encryption[:tls_options][:verify_mode] != OpenSSL::SSL::VERIFY_NONE
+        if encryption && encryption[:tls_options] &&
```

Comment:
Perhaps `#encryption` should always return a hash, even if no encryption is specified. This nil check feels awkward.